pyregexp/engine.py

```diff
@@ -59,6 +65,10 @@ def return_fnc(res: bool, str_i: int, all_matches: list, return_matches: bool) -
             else:
                 return res, str_i
 
+        if ignore_case:
+           re = unicodedata.normalize("NFKD", re).casefold()
+           string = unicodedata.normalize("NFKD", string).casefold()
```

Comment:
Probably casefolding is a bit too aggressive as the only option.
It would be better to have an option for "classic" ignore case, and one for casefolding